README.md

```diff
@@ -1 +1,83 @@
-# CytoPipe
\ No newline at end of file
+# Table of contents
+
+- [Table of contents](#table-of-contents)
+- [CytoPipe](#cytopipe)
+- [Formatting](#formatting)
+  - [smkfmt](#smkfmt)
+- [Testing](#testing)
+  - [Dry Runs](#dry-runs)
+
+# CytoPipe
+
+# Formatting
+
+## smkfmt
+
+`snakefmt` is a `snakemake` code formatter that is developed and maintained by the developers of `snakemake`.
+
+In order to use `snakefmt`, `snakemake` must be installed, which can be found here: [snakemake-installation](https://snakemake.readthedocs.io/en/stable/getting_started/installation.html)
+
+`snakefmt` installation processes can be found in the [snakefmt](https://github.com/snakemake/snakefmt#install) repository.
+
+# Testing
+
+## Dry Runs
+
+`Snakemake`'s `dry-runs` is a simple test that ensures that all the required inputs and expected outputs are satisfied. No execution is conducted therefore any potential errors found within the executables **will not be captured.**
+
+Below is an example on how to conduct a `dry-run` on `snakemake`:
+
+```text
+snakemake -n
+```
+
+Using the `-n`, by default, will conduct a `dry-run` to all rules that exists within the workflow
+
+One can specify a specific rule to conduct a `dry-run` test by inputting the name of the rule after the `-n` flag . This will only check all the required inputs and expected outputs that are specific to the specified rule.
```

Comment:
```suggestion
One can specify a specific rule to conduct a `dry-run` test by inputting the name of the rule after the `-n` flag.
This will only check all the required inputs and expected outputs that are specific to the specified rule.
```

Neat!